tensorflow/core/tpu/tpu_initializer_helper.cc

```diff
@@ -51,6 +55,55 @@ bool GetEnvBool(const char* name, bool defval) {
 
 }  // namespace
 
+// This function gets pid of a process and checks if that process is using tpu.
+// It is not able to check processes that are owned by another user.
+bool IsTpuUsed(long pid) {
+  std::string path = absl::StrCat("/proc/", pid, "/fd");
+  DIR* raw_fd_dir = opendir(path.c_str());
+  if (!raw_fd_dir) {
+    return false;
+  }
+  std::unique_ptr<DIR, int (*)(DIR*)> fd_dir(raw_fd_dir, closedir);
+  struct dirent* ent;
+  char line[100];
+  while (ent = readdir(raw_fd_dir)) {
+    if (!isdigit(*ent->d_name)) continue;
+    long fd = strtol(ent->d_name, NULL, 10);
+    path = absl::StrCat("/proc/", pid, "/fd/", fd);
+    if (!readlink(path.c_str(), line, 100)) return false;
+    if (strncmp(line, "/dev/accel0", 11) != 0) continue;
+    return true;
+  }
+  return false;
+}
+
+// This function iterates through all the processes in /proc and logs if any
+// process it was able to check is using the TPU. It does not have permission to
+// processes owned by another user. todo (shahrokhi) use
```

Comment:
```suggestion
// processes owned by another user.
// TODO(shahrokhi): ...
```
nit: format TODO like this (capitalize, put on separate line)